pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -80,19 +158,134 @@ func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName
 			if pkgShortName == "tkg-storageclass" {
 				continue
 			}
-			verifyPackageInstall(ctx, wccl, addonutil.GeneratePackageInstallName(wcClusterName, pkgShortName), pkgName, pkgVersion)
+			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: constants.TkgNamespace},
+				Spec: kapppkgv1alpha1.PackageSpec{RefName: pkgName, Version: pkgVersion}})
 		}
 	}
 
+	return packages, nil
+}
+
+// checkClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
+func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
+	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
+
+	var clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap
+	if isManagementCluster {
+		clusterBootstrap = getClusterBootstrap(ctx, mccl, mcClusterNamespace, mcClusterName)
+	} else {
+		clusterBootstrap = getClusterBootstrap(ctx, mccl, wcClusterNamespace, wcClusterName)
```

Comment:
do we have to get clusterbootstrap from workload cluster?

Reply:
No, true that the resource exists on the management cluster itself, but it is under each cluster's name/namespace anyway